pandas/_libs/groupby.pyx

```diff
@@ -261,23 +280,41 @@ def group_cumsum(
             for j in range(K):
                 val = values[i, j]
 
-                isna_entry = _treat_as_na(val, is_datetimelike)
+                if uses_mask:
```

Comment:
Rather than continue `if uses_mask:` checks we make that the outermost branch? Might help readability to keep the logic in two different branches rather than continued checks within one

Reply:
I tried that, but imo that reduces readability. The uses_mask is a simple if_else branch, if we move this outside, it is hard to see that the actual logic of the algorithm is the same in both branches (and we have to keep it consistent over time).